packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -120,21 +101,69 @@ export function trackViews(location: Location, lifeCycle: LifeCycle) {
     return newView(lifeCycle, location, isRecording, ViewLoadingType.ROUTE_CHANGE, currentView.url, startClocks, name)
   }
 
+  function startViewLifeCycle() {
+    lifeCycle.subscribe(LifeCycleEventType.SESSION_RENEWED, () => {
+      // do not trigger view update to avoid wrong data
+      currentView.end()
+      if (areViewsTrackedAutomatically) {
+        // Renew view on session renewal
+        currentView = trackViewChange()
+      }
+    })
+
+    // End the current view on page unload
+    lifeCycle.subscribe(LifeCycleEventType.BEFORE_UNLOAD, () => {
+      currentView.end()
+      currentView.triggerUpdate()
+    })
+
+    lifeCycle.subscribe(LifeCycleEventType.RECORD_STARTED, () => {
+      isRecording = true
+      currentView.updateHasReplay(true)
+    })
+
+    lifeCycle.subscribe(LifeCycleEventType.RECORD_STOPPED, () => {
+      isRecording = false
+    })
```

Comment:
We probably want to prevent recording to start if there is no view, like other public APIs, else this will not be accurate.

Reply:
I look into it